### DISCUSSION TITLE
Get Logged In User in Data Objects

### DISCUSSION THREAD

**Discussion by ramesh081** [pimcore/pimcore#10219] [General]
Hi, I am new to pimcore development. I have logged as admin and creating a new data object class. I want to show logged in user as default value in data object Textbox. What i have to do ?
Please guide.

**Comment by NiklasBr** [comment]
Because the User reference field does not support _default value generator_ service (not clear why):

![image](https://user-images.githubusercontent.com/279826/132364230-61986c06-1a79-4931-9d09-155f3c5a7d1d.png)

You would probably have to [add an Event Listener](https://pimcore.com/docs/pimcore/current/Development_Documentation/Extending_Pimcore/Event_API_and_Event_Manager.html) where you hook into:

https://github.com/pimcore/pimcore/blob/7edef74976d037e913a6ea7d2c497a566d62db57/lib/Event/DataObjectEvents.php#L20-L25

or

https://github.com/pimcore/pimcore/blob/7edef74976d037e913a6ea7d2c497a566d62db57/lib/Event/DataObjectEvents.php#L44-L52
